test/e2e/network/network_policy.go

```diff
@@ -36,290 +38,247 @@ are correctly enforced by a networking plugin. It accomplishes this by launching
 a simple netcat server, and two clients with different
 attributes. Each test case creates a network policy which should only allow
 connections from one of the clients. The test then asserts that the clients
-failed or succesfully connected as expected.
+failed or successfully connected as expected.
 */
 
 var _ = SIGDescribe("NetworkPolicy", func() {
+	var service *v1.Service
+	var podServer *v1.Pod
 	f := framework.NewDefaultFramework("network-policy")
 
-	It("should support a 'default-deny' policy [Feature:NetworkPolicy]", func() {
-		ns := f.Namespace
-
-		By("Create a simple server.")
-		podServer, service := createServerPodAndService(f, ns, "server", []int{80})
-		defer cleanupServerPodAndService(f, podServer, service)
-		framework.Logf("Waiting for Server to come up.")
-		err := framework.WaitForPodRunningInNamespace(f.ClientSet, podServer)
-		Expect(err).NotTo(HaveOccurred())
-
-		// Create a pod with name 'client-can-connect', which should be able to communicate with server.
-		By("Creating client which will be able to contact the server since no policies are present.")
-		testCanConnect(f, ns, "client-can-connect", service, 80)
-
-		By("Creating a network policy denying all traffic.")
-		policy := &networking.NetworkPolicy{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "deny-all",
-			},
-			Spec: networking.NetworkPolicySpec{
-				PodSelector: metav1.LabelSelector{},
-				Ingress:     []networking.NetworkPolicyIngressRule{},
-			},
-		}
-
-		policy, err = f.InternalClientset.Networking().NetworkPolicies(ns.Name).Create(policy)
-		Expect(err).NotTo(HaveOccurred())
-		defer cleanupNetworkPolicy(f, policy)
+	Context("NetworkPolicy between server and client", func() {
+		BeforeEach(func() {
+			By("Creating a simple server that serves on port 80 and 81.")
+			podServer, service = createServerPodAndService(f, f.Namespace, "server", []int{80, 81})
+			framework.Logf("Waiting for Server to come up.")
+			err := framework.WaitForPodRunningInNamespace(f.ClientSet, podServer)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Wait for the server pods to become ready.
+			for !(podServer.Status.ContainerStatuses[0].Ready && podServer.Status.ContainerStatuses[1].Ready) {
+				time.Sleep(1 * time.Second)
```

Comment:
This won't work; you need to re-fetch the Pod object from the server. The existing Pod object that you have won't change.

But I'm pretty sure there are already "wait for a pod to be ready" utility functions somewhere in test/e2e